src/SprykerEco/Yves/Adyen/Plugin/Router/AdyenRouteProviderPlugin.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Yves\Adyen\Plugin\Router;
+
+use Spryker\Yves\Router\Plugin\RouteProvider\AbstractRouteProviderPlugin;
+use Spryker\Yves\Router\Route\RouteCollection;
+
+class AdyenRouteProviderPlugin extends AbstractRouteProviderPlugin
+{
+    protected const BUNDLE_NAME = 'Adyen';
+    protected const CALLBACK_CONTROLLER_NAME = 'Callback';
+    protected const NOTIFICATION_CONTROLLER_NAME = 'Notification';
+
+    /**
+     * @param \Spryker\Yves\Router\Route\RouteCollection $routeCollection
+     *
+     * @return \Spryker\Yves\Router\Route\RouteCollection
+     */
+    public function addRoutes(RouteCollection $routeCollection): RouteCollection
+    {
+        $routeCollection = $this->addAdyenNotificationRoute($routeCollection);
+        $routeCollection = $this->addAdyenRedirectSofortRoute($routeCollection);
+        $routeCollection = $this->addAdyenRedirectCard3dRoute($routeCollection);
+        $routeCollection = $this->addAdyenRedirectIdealRoute($routeCollection);
+        $routeCollection = $this->addAdyenRedirectPaypalRoute($routeCollection);
+        $routeCollection = $this->addAdyenRedirectAlipayRoute($routeCollection);
+        $routeCollection = $this->addAdyenRedirectWechatPayRoute($routeCollection);
+
+        return $routeCollection;
+    }
+
+    /**
+     * @param \Spryker\Yves\Router\Route\RouteCollection $routeCollection
+     *
+     * @return \Spryker\Yves\Router\Route\RouteCollection
+     */
+    protected function addAdyenNotificationRoute(RouteCollection $routeCollection): RouteCollection
+    {
+        $routeCollection->add(
+            'adyen-notification',
+            $this->buildRoute('/adyen/notification', static::BUNDLE_NAME, static::NOTIFICATION_CONTROLLER_NAME, 'index')
```

Comment:
Only post allowed here.
Probably we need to add allowed methods for each route.